### PR TITLE
Fix formatting of more date references

### PR DESCRIPTION
**Pull request by camelid**
It turns out I missed several in #1066.

### CODE REVIEW

File: src/type-inference.md [hunk 1]
```diff
@@ -71,8 +71,8 @@ inference works, or perhaps this blog post on
 
 [Unification in the Chalk project]: http://smallcultfollowing.com/babysteps/blog/2017/03/25/unification-in-chalk-part-1/
 
-All told, the inference context stores four kinds of inference variables (as of
-<!-- date: 2018-01 --> January 2018):
+All told, the inference context stores four kinds of inference variables
+(as of <!-- date: 2018-01 --> January 2018):
```
Comment (camelid) [inline src/type-inference.md:75]: It turns out that mdBook freaks out when a line begins with an HTML comment:

image

Not sure if it's a bug or expected behavior.

Comment (camelid) [reply src/type-inference.md:75]: It generates this HTML:

```html
<p>All told, the inference context stores four kinds of inference variables (as of</p>
<!-- date: 2018-01 --> January 2018):
```

Probably pulldown-cmark is the one doing that.

**Review by JohnTitor — APPROVED** [review]
Seems you still missed the below:

https://github.com/rust-lang/rustc-dev-guide/blob/c4c0f33a1d1bfad621b15b9dd797d0ed25d70ac1/src/tests/intro.md#L129-L130

Otherwise LGTM (I wonder if we could somehow detect with date-check 🤔).

**Comment by camelid** [comment]
> Seems you still missed the below:
> 
> https://github.com/rust-lang/rustc-dev-guide/blob/c4c0f33a1d1bfad621b15b9dd797d0ed25d70ac1/src/tests/intro.md#L129-L130
> 
> Otherwise LGTM (I wonder if we could somehow detect with date-check 🤔).

Great catch, thanks!

**Pull request merged**